src/patterns/menu/Menu.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useRef, useState } from 'react';
+import type { Property } from 'csstype';
+
+import MenuItem from '../../components/menu-item';
+import Icon from '../../foundations/icon';
+import Shape from '../../foundations/shape';
+import Popover from '../../utilities/popover';
+import Grid from '../../utilities/grid';
+
+import MenuRoot, { MenuLabelRoot } from './Menu.root';
+import type { PSubmenu } from './Menu.props';
```

Comment:
nit: I like to order them the other way around: 
1. default imports
2. named imports

Its a shame that you have to split type imports like this :/